app/torii-adapters/arcgis-oauth-bearer.js

```diff
@@ -35,77 +35,92 @@ export default Ember.Object.extend({
   /**
    * Promisified getJson
    */
-  _getJson (url) {
-    return new Ember.RSVP.Promise(function (resolve, reject) {
-      Ember.$.ajax({
-        url: url,
-        dataType: 'json',
-        success: Ember.run.bind(null, function (data) {
-          if (data.error) {
-            Ember.debug('torii:adapter:arcgis-oauth-bearer:open portals/self call shows token was not valid.');
-            reject(data);
-          } else {
-            resolve(data);
-          }
-        }),
-        error: Ember.run.bind(null, reject)
-      });
-    });
-  },
+  // _getJson (url) {
+  //   return new Ember.RSVP.Promise(function (resolve, reject) {
+  //     Ember.$.ajax({
+  //       url: url,
+  //       dataType: 'json',
+  //       success: Ember.run.bind(null, function (data) {
+  //         if (data.error) {
+  //           Ember.debug('torii:adapter:arcgis-oauth-bearer:open portals/self call shows token was not valid.');
+  //           reject(data);
+  //         } else {
+  //           resolve(data);
+  //         }
+  //       }),
+  //       error: Ember.run.bind(null, reject)
+  //     });
+  //   });
+  // },
 
   /**
    * Open a session by fetching portal/self from
    * the portal
    */
   open (authentication) {
-    let token = authentication.authorizationToken.token;
-    let expires = Date.now() + (authentication.authorizationToken.expires_in * 1000); // seconds from now
+    // TODO?: If we have a cookie but the token is invalid (i.e. for a different portal)
+    // then this call will return a 499-in-a-200.
 
-    // Ember.debug('torii:adapter:arcgis-oauth-bearer:open token...' + token);
-    let portalSelfUrl = this.get('portalBaseUrl') + '/sharing/rest/portals/self?f=json&token=' + token;
+    // instantiate an auth session from what's in the cookie
+    if (!authentication.session) {
+      authentication.session = arcgisRest.UserSession.deserialize(authentication.authorizationToken.serializedSession);
+    }
+
+    // session is hydrated with the portal info and token
+    return arcgisRest.getSelf({
+      authentication: authentication.session
+    })
+      .then((portal) => {
+        Ember.debug('torii:adapter:arcgis-oauth-bearer:open got response from portal/self & assigning to session');
+
+        if (ENV.torii.providers['arcgis-oauth-bearer'].loadGroups) {
+          // make a request to get user's groups
+          let username = portal.user.username;
+
+          const userUrl = arcgisRest.getPortalUrl({
+            authentication: authentication.session
+          }) + `/community/users/${username}`
+
+          return arcgisRest.request(userUrl, {
```

Comment:
Just curious, would this use the new `getUser()` in https://github.com/Esri/arcgis-rest-js/pull/170 ?

Reply:
that's definitely the short/medium-term plan, but not in this PR.